website_sale_one_step_checkout/controllers/main.py

```diff
@@ -0,0 +1,277 @@
+                'zip': post['shipping_zip'],
+                'street': post['shipping_street'] + ' ' + post.get(
+                    'shipping_street_number'),
+                'street_name': post['shipping_street'],
+                'street_number': post['shipping_street_number'],
+                'city': post['shipping_city'],
+                'name': post['shipping_name'],
+                'email': post['email'],
+                'type': 'delivery',
+                'parent_id': partner.id,
+                'country_id': post['shipping_country_id'],
+                'state_id': post['shipping_state_id'],
+            }
+            domain = [
+                (key, '_id' in key and '=' or 'ilike',
+                 '_id' in key and value and int(value) or value)
+                for key, value in shipping_info.items() if key in
+                self.mandatory_billing_fields + ['type', 'parent_id']]
+            shipping_partners = orm_partner.sudo().search(domain)
+            if shipping_partners:
+                shipping_partner = shipping_partners[0]
+                shipping_partner.write(shipping_info)
+            else:
+                shipping_partner = orm_partner.sudo().create(shipping_info)
+
+        order_info = {
+            'partner_id': partner.id,
+            'message_follower_ids': [(4, partner.id),
+                                     (3, request.website.partner_id.id)],
+            'partner_invoice_id': partner.id
+        }
+        order_info.update(
+            request.env['sale.order'].sudo().onchange_partner_id(
+                partner.id)['value'])
+        # we need to update partner_shipping_id after onchange_partner_id()
+        # call otherwise the deselection of the option 'Ship to a different
+        # address'  would be overwritten by an existing shipping partner type
+        order_info.update({
+            'partner_shipping_id': (shipping_partner and
+                                    shipping_partner.id) or partner.id})
+        order_info.pop('user_id')
+
+        order.sudo().write(order_info)
+        request.session['sale_last_order_id'] = order.id
+        return {'success': True}
+
+    def do_change_delivery(self, order, carrier_id):
+        """Apply delivery amount to current sale order."""
+        if not order or not carrier_id:
+            return {'success': False}
+
+        # order_id is needed to get delivery carrier price
+        if not request.context.get('order_id'):
+            request.context['order_id'] = order.id
+
+        # recompute delivery costs
+        request.env['sale.order']._check_carrier_quotation(
+            order, force_carrier_id=carrier_id)
+
+        # generate updated total prices
+        updated_order = request.website.sale_get_order()
+
+        rml_obj = report_sxw.rml_parse(request.cr, SUPERUSER_ID,
+                                       request.env['product.product']._name,
+                                       context=request.context)
```

Comment:
leading rml_obj just to format price looks strange, but i don't know any better solution